### PR TITLE
raidboss: p7s inviolate bonds, inviolate purgation

### PR DESCRIPTION
**Pull request by kissorjeyabalan**
Quick early implementation for inviolate bonds and inviolate purgation. Seems to be working as intended - might need to test a bit more. The duration is set to match the last debuffs expiring - not sure if too long.

### CODE REVIEW

File: ui/raidboss/data/06-ew/raid/p7s.ts [hunk 1]
```diff
@@ -25,6 +46,90 @@ const triggerSet: TriggerSet<Data> = {
         text: 'Split Tankbusters',
       },
     },
+    {
+      id: 'P7S Spark of Life',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '7839', source: 'Agdistis', capture: false }),
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: 'Raidwide + Bleed',
```
Comment (quisquous) [inline ui/raidboss/data/06-ew/raid/p7s.ts:55]:
```suggestion
        text: 'aoe + bleed',
```

Mostly just to match other examples of aoe+ bleed elsewhere.  I know people say "raidwide", but cactbot says "aoe" for all of these, and it's better to be consistent.

File: ui/raidboss/data/06-ew/raid/p7s.ts [hunk 2]
```diff
@@ -25,6 +46,90 @@ const triggerSet: TriggerSet<Data> = {
...
+      },
+    },
+    {
+      id: 'P7S Inviolate Bonds',
+      type: 'GainsEffect',
+      // CEC/D45 = Inviolate Winds
+      // CED/D56 = Holy Bonds
+      netRegex: NetRegexes.gainsEffect({ effectId: ['CEC', 'D45'] }),
+      condition: (data, matches) => matches.target === data.me,
```
[the remark this answers is not in the document]
Comment (JLGarber) [reply ui/raidboss/data/06-ew/raid/p7s.ts:64]: In this case it should be just a bare `condition: Conditions.targetIsYou(),`

[the remark this answers is not in the document]
Comment (panicstevenson) [reply ui/raidboss/data/06-ew/raid/p7s.ts:64]: You're right, oops. Updated suggestion to reflect that for ease-of-button-press.

**Comment by kissorjeyabalan** [comment]
@panicstevenson @quisquous Hey, thanks. I've done the suggested changes. Everything seems to be fine in the emulator with a few log files I had.

**Comment by quisquous** [comment]
Thanks for the trigger.  Worked fine for me!

I think if you (or others) were interested in other changes to make these even better:
(1) the Inviolate Bonds could say "stack first" and then with some delay could say "spread" rather than keeping it on screen the whole time
(2) similarly, the Inviolate Purgation could say stack or spread after each hit to remind you where to go for each one (in case you lose count)
(3) track where the safe spots were and say "stack NW" or "spread S" sort of thing

**Pull request merged**